Pod/Classes/WPMediaPickerViewController.m

```diff
@@ -1445,35 +1465,18 @@ - (void)keyboardWillHideNotification:(NSNotification *)notification
 - (void)centerEmptyView
 {
     if (self.emptyViewController) {
-        CGRect emptyViewFrame = [self getEmptyViewFrame];
-
-        if ([self.searchBar.text isEqualToString:@""]) {
-            emptyViewFrame.origin.y -= self.searchBar.frame.size.height/2;
-        }
-        emptyViewFrame.size.height -= self.view.layoutMargins.bottom;
-        emptyViewFrame.size.height -= self.view.layoutMargins.top;
-        _emptyViewController.view.frame = emptyViewFrame;
+        [self.view layoutIfNeeded];
     } else {
         self.emptyView.center = self.collectionView.center;
         self.emptyView.frame = [self getEmptyViewFrame];
-    }
+   }
 }
 
 - (CGRect)getEmptyViewFrame
 {
-    CGRect emptyViewFrame;
-
-    if (_emptyViewController) {
-        emptyViewFrame = self.collectionView.frame;
-    } else {
-        emptyViewFrame = self.emptyView.frame;
-    }
-
-    CGFloat superviewHeight = self.collectionView.frame.size.height;
-    CGFloat totalInsets = self.collectionView.contentInset.top + self.collectionView.contentInset.bottom;
+    CGRect emptyViewFrame = self.emptyView.frame;
```

Comment:
I see we still do some frame manipulation when the delegate doesn't provide an empty view controller and we fall back on using `emptyView`.

What do you think of adding a "container" view? If we go that route, we'll add constraints to the container view and hide/show it as needed. And we'll embed in it, either the view provided by the delegate or the default view. This way we unify the logic between the two possible paths. WDYT?

Reply:
I think is feasible to do it 😄, I decided to keep this doesn't have a side effect in another area, and to not change something that is already done on the code, but let me prepare some code with this approach in the next commit. Thanks for this!

Reply:
Change made here https://github.com/wordpress-mobile/MediaPicker-iOS/pull/399/commits/ded72ca809b30d35e75f2c6213e4be8951586531 😄 

![emptyView](https://user-images.githubusercontent.com/20734091/198411009-c45a671f-7b98-423a-80e4-43e400934802.gif)